src/v/pandaproxy/probe.h

```diff
@@ -12,6 +12,7 @@
 #pragma once
 
 #include "metrics/metrics.h"
+#include "pandaproxy/server.h"
```

Comment:
I think `server::ctx` can be forward declared.

Reply:
`server::context_t` cannot be forward declared because it is an internal struct,.
Given how it is inherited from in the `ctx_server::context_t` and indirectly in the `auth_ctx_server::context_t`,  I don't think it is worth bringing out of the `server` just to break a compilation dependency.

Reply:
Ahh, nothing to do here for this PR, then.